xplique/commons/nlp.py

```diff
@@ -0,0 +1,288 @@
+    clause_type
+        A list with the types of clauses to keep. Each clause shall be a string
+        corresponding to one of Spacy Tag dictionnary.
+        Example: clause_type = ['NNP', 'VBZ']
+        If None, all clauses are kept.
+    """
+
+    def __init__(self,
+                 pipeline: Type[Language],
+                 clause_type: List[str] = None):
+        super().__init__(language=None)
+        self.clause_type = clause_type
+        self.pipeline = pipeline
+
+    def extract_tokens(self, sentence: Union[List[str], str]) -> Tuple[List[str], str]:
+        """
+        Separates the input texts into clauses, and only keeps the ones belonging to
+        the specified types.
+        If clause_type is None, the texts are split but all the clauses are kept.
+
+        Parameters
+        ----------
+        sentence
+            A list of strings that we wish to separate into clauses.
+
+        Returns
+        -------
+        clause_list
+            A list with input texts split into clauses.
+        separator
+            The separator used for spliting the sentence.
+        """
+        if not isinstance(sentence, str):
+            sentence = '.'.join(sentence)
+
+        clause_list = []
+        for token in self.pipeline(sentence):
+            if self.clause_type is None:
+                clause_list.append(token.text)
+            elif token.tag_ in self.clause_type:
+                clause_list.append(token.text)
+
+        return clause_list, self.separator
+
+
+class ExtractorFactory():
+    """
+    Factory for extractor classes.
+    """
+    @staticmethod
+    def get_extractor(extract_fct="sentence", tagger=None, pipeline=None, clause_type=None):
+        """
+        Get an instance of an extractor based on the specified extraction function.
+
+        Parameters
+        ----------
+        extract_fct
+            The type of extraction function to use, either 'word', 'sentence', 'excerpt',
+            'flair_clause', 'spacy_clause'.
+            Default is "sentence".
+        tagger
+            A Flair SequenceTagger to use for Flair clause extractor.
+        pipeline
+            A Spacy Pipeline to use for Spacy clause extractor.
+        clause_type
+            Additional parameter specifying the type of clause (default is None).
+
+        Returns
+        -------
+        Extractor
+            An instance of the selected extractor based on the specified function.
+        """
+        if extract_fct == "flair_clause":
+            return FlairClauseExtractor(tagger, clause_type)
+        if extract_fct == "spacy_clause":
+            return SpacyClauseExtractor(pipeline, clause_type)
+        if extract_fct == "sentence":
+            return SentenceExtractor()
+        if extract_fct == "word":
+            return WordExtractor()
+        if extract_fct == "excerpt":
+            return ExcerptExtractor()
+        raise ValueError("Extraction function can be only 'clause', \
```

Comment:
You did not include `"flaire_clause"` and `"spacy_clause"` in the error message, but you left `"clause"` which is not a possibility.